### PR TITLE
Buffer overflow fix

### PR DESCRIPTION
**Pull request by Ajaja**
Details:
https://forum.farmanager.com/viewtopic.php?p=158555#p158555

### CODE REVIEW

**Comment by johnd0e** [comment]
The commit message is bad.

File: plugins/multiarc/custom.cpp [hunk 1]
```diff
@@ -989,9 +989,9 @@ void ParseListingItemRegExp(Match match,
 {
 
     if(const char *p = match["name"])
-        lstrcat(Item->FindData.cFileName, p);
+        lstrcpyn(Item->FindData.cFileName, p, sizeof(Item->FindData.cFileName));
```
Comment (alabuzhev) [inline plugins/multiarc/custom.cpp]: Why the original code was appending the strings rather than copying them?

Comment (Ajaja) [reply plugins/multiarc/custom.cpp]: Item - this structure must be filled:
https://github.com/FarGroup/FarManager/blob/86c26a80a21e24e3c2a7b4d259a76d3ef34d44fc/plugins/multiarc/README.TXT#L65

Comment (Ajaja) [reply plugins/multiarc/custom.cpp]: I think it always should be empty
https://github.com/FarGroup/FarManager/blob/86c26a80a21e24e3c2a7b4d259a76d3ef34d44fc/plugins/multiarc/arcread.cpp#L97

**Pull request merged**